File: BaseLibrary/src/main/java/com/pengyu/base/utils/fingerprint/MyAuthCallback.java
package com.pengyu.base.utils.fingerprint;

import android.support.v4.hardware.fingerprint.FingerprintManagerCompat;

public class MyAuthCallback extends FingerprintManagerCompat.AuthenticationCallback {

    private FingerUtil.IFingerprintResultListener iFingerprintResultListener;

    MyAuthCallback(FingerUtil.IFingerprintResultListener iFingerprintResultListener) {
        super();
        this.iFingerprintResultListener = iFingerprintResultListener;
    }

    /**
     * 验证错误信息
     */
    @Override
    public void onAuthenticationError(int errMsgId, CharSequence errString) {
        super.onAuthenticationError(errMsgId, errString);
        if (iFingerprintResultListener != null) {
            iFingerprintResultListener.onAuthenticateError();
        }
    }

    /**
     * 身份验证帮助
     */
    @Override
    public void onAuthenticationHelp(int helpMsgId, CharSequence helpString) {
        super.onAuthenticationHelp(helpMsgId, helpString);
    }

    /**
     * 验证成功
     */
    @Override
    public void onAuthenticationSucceeded(FingerprintManagerCompat.AuthenticationResult result) {
        super.onAuthenticationSucceeded(result);
        if (iFingerprintResultListener != null) {
            iFingerprintResultListener.onAuthenticateSuccess();
        }
    }

    /**
     * 验证失败
     */
    @Override
    public void onAuthenticationFailed() {
        super.onAuthenticationFailed();
        if (iFingerprintResultListener != null) {
            iFingerprintResultListener.onAuthenticateFailed();
        }
    }
}

File: BaseLibrary/src/main/java/com/pengyu/base/utils/fingerprint/MyAuthCallback.kt
package com.pengyu.base.utils.fingerprint

import android.support.v4.hardware.fingerprint.FingerprintManagerCompat

class MyAuthCallback internal constructor(private val iFingerprintResultListener: FingerUtil.IFingerprintResultListener?) :
    FingerprintManagerCompat.AuthenticationCallback() {

    /**
     * 验证错误信息
     */
    override fun onAuthenticationError(errMsgId: Int, errString: CharSequence?) {
        super.onAuthenticationError(errMsgId, errString)
        iFingerprintResultListener?.onAuthenticateError()
    }

    /**
     * 身份验证帮助
     */
    override fun onAuthenticationHelp(helpMsgId: Int, helpString: CharSequence?) {
        super.onAuthenticationHelp(helpMsgId, helpString)
    }

    /**
     * 验证成功
     */
    override fun onAuthenticationSucceeded(result: FingerprintManagerCompat.AuthenticationResult?) {
        super.onAuthenticationSucceeded(result)
        iFingerprintResultListener?.onAuthenticateSuccess()
    }

    /**
     * 验证失败
     */
    override fun onAuthenticationFailed() {
        super.onAuthenticationFailed()
        iFingerprintResultListener?.onAuthenticateFailed()
    }
}
